src/sessions.ts

```diff
@@ -555,33 +553,33 @@ function isMaxTimeMSExpiredError(err: MongoError) {
   );
 }
 
-function attemptTransactionCommit<T>(
+async function attemptTransactionCommit<T>(
   session: ClientSession,
   startTime: number,
   fn: WithTransactionCallback<T>,
   result: any,
```

Comment:
```suggestion
  result: T,
```

This does not matter too much, but it technically more correct